src/api/routes/ws/socket-io.ts

```diff
@@ -17,8 +17,39 @@ import {
   getTxFromDataStore,
   parseDbTx,
 } from '../../controllers/db-controller';
-import { isProdEnv, logError, logger } from '../../../helpers';
+import { isProdEnv, isValidPrincipal, logError, logger } from '../../../helpers';
 import { WebSocketPrometheus } from './metrics';
+import { isValidTxId } from '../../../api/query-helpers';
+
+function getInvalidSub(subscriptions: Topic | Topic[]): undefined | string {
```

Comment:
```suggestion
function getInvalidSubscriptionTopics(subscriptions: Topic | Topic[]): undefined | Topic[] {
```

Reply:
The return type should remain `string` as this function returns `string[]` that is not `Topic[]`.